### ISSUE TITLE
Editing is really slow

### ISSUE THREAD

**Issue by PierrunoYT** [cline/cline#744]
### Which API Provider are you using?

OpenRouter

### Which Model are you using?

Sonnet 3.5

### What happened?

I see that the diff applying and editing is really slow and needs improvment 

### Steps to reproduce

I see that the diff applying and editing is really slow and needs improvment 

### Additional context

I see that the diff applying and editing is really slow and needs improvment

**Comment by Astlaan** [comment]
I can concur. It is very slow because it literally seeks through the whole file, even if it only has to change one line.
It would be nicer to use a strategy similar to aider: it generates SEARCH/REPLACE blocks, which find the code section and apply the changes "instantly" without having to go through the whole file.

**Comment by PierrunoYT** [comment]
> I can concur. It is very slow because it literally seeks through the whole file, even if it only has to change one line. It would be nicer to use a strategy similar to aider: it generates SEARCH/REPLACE blocks, which find the code section and apply the changes "instantly" without having to go through the whole file.

Can you check my PR?

**Comment by CiberNin** [comment]
This is intended behavior that won't be fixed until Anthropic released quick edit API.

**Comment by saoudrizwan** [comment]
OpenAI just released a new API featured called [predicted outputs](https://platform.openai.com/docs/guides/latency-optimization#use-predicted-outputs) that would fix this issue entirely. It essentially spits out the tokens its read before and only has to "think" about new tokens (the new edits to the file), making the edit faster and more reliable without the code truncation. The issue is that this API is very rudimentary and limited (doesn't allow images, only works with gpt4o, etc) so it doesn't fit as nicely into Cline as I'd like. I'm hoping Anthropic comes out with their version of this API soon (which they've pre-released in Zed, called [Fast Edit Mode](https://www.youtube.com/watch?v=6CCuyTMhZKw)), so I can add it to Cline ASAP and fix this problem.

Pls join the [discord](https://discord.gg/cline) for updates about this, and thank you for bearing with me in the meantime 🙏

**Comment by msTokens** [comment]
still slow :D to this day

**Comment by CiberNin** [comment]
Yeah this should never have been closed I think said thought we were saying the AI is slow but no the bottleneck is literally the animation of the changes. Even if only one like in a ten thousand line file is changed.

I've verified this experimentally by commenting out that animation entirely. Especially bad because sometimes the animation randomly goes super slow even though the result stream speed has no issues.

**Comment by pavlelee** [comment]
> Yeah this should never have been closed I think said thought we were saying the AI is slow but no the bottleneck is literally the animation of the changes. Even if only one like in a ten thousand line file is changed.
> 
> I've verified this experimentally by commenting out that animation entirely. Especially bad because sometimes the animation randomly goes super slow even though the result stream speed has no issues.

This issue greatly affects efficiency, and should be reopened.

**Comment by dcbartlett** [comment]
Reopening as this is actively being worked in https://github.com/cline/cline/pull/1611

**Comment by pjongeneel** [comment]
Why did 1611 get closed?

**Comment by ZionNFT** [comment]
yep, what's happening re. this? It's basically making Cline unusable atm

**Comment by dancingmadkefka** [comment]
Ya this issue is pretty weird - it's being done on purpose right? It's like a "feature" or what? I don't get it.

**Comment by ulivz** [comment]
See: https://github.com/kortix-ai/fast-apply
